fili-core/src/main/java/com/yahoo/bard/webservice/servlet/RoleImpersonationFilter.java

```diff
@@ -0,0 +1,220 @@
+// Copyright 2022 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.servlet;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.inject.Singleton;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+
+/**
+ * Use the 'From' header and a service to build a user Principal based on a service provider.
+ */
+@Singleton
+public abstract class RoleImpersonationFilter implements ContainerRequestFilter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RoleImpersonationFilter.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    /**
+     * Property to allow translation of roles from distinct role domains into a common nomenclature.
+     */
+    public static final String ROLE_NAME_ALIAS_KEY = SYSTEM_CONFIG.getPackageVariableName("role_name_aliases");
+
+    /**
+     * The expected structure of a From header is userName@domain.domain.
+     */
+    private static final Pattern EMAIL_PATTERN = Pattern.compile("(\\w+)@([a-zA-Z.])");
+
+    /**
+     * The map of incoming roles to expected system roles.  This mapping allows reconciliation of role sources
+     * with difference casing, separators, etc.
+     */
+    private final Map<String, String> aliases;
+
+    /**
+     * Constructor. Builds the mapping of aliases based on system config.
+     *
+     */
+    public RoleImpersonationFilter() {
+        List<String> roleAliases = SYSTEM_CONFIG.getListProperty(ROLE_NAME_ALIAS_KEY);
+        Map<String, String> inProgressAliases = new HashMap<>();
+
+        if (!roleAliases.isEmpty()) {
+            for (String aliasMapping : roleAliases) {
+                String[] mapping = aliasMapping.split(":");
+                if (mapping.length != 2) {
+                    LOG.warn(
```

Comment:
I would just have this throw an exception and let the system crash. It happens at start up time, so it's not like this will crash anything while running, and it ensures that if there is a malformed role mapping, developers are aware of it. What's going on with role-based stuff isn't always obvious in a running system, and we don't want to be in a situation where the system is silently mishandling potentially important roles because of a dumb typo that we caught, but by and large ignored.